### DISCUSSION TITLE
No RenderDevice in RenderApp's World

### DISCUSSION THREAD

**Discussion by dragostis** [bevyengine/bevy#10028] [Q&A] [answered]
Is there any reason why the `RenderApp`'s `World` doesn't have access to the `RenderDevice` (and `RenderQueue`)? If you have a render graph `Node` that would like to cache a texture/buffer, this can't be created in `FromWorld`.

Is there any other way to access these resources when creating the node?

Specifically, the `World` in `FromWorld` that gets triggered when adding the node with `render_app.add_render_graph_node::<MyNode>`.

**Comment by IceSentry** [comment, marked as answer]
The RenderWorld does have access to it and in fact it's the only place where they are accessible. The issue is most likely that you are calling the FromWorld impl in the plugin build() function. What you need to do is call it in the finish() function. 

Since bevy 0.11 we now initialize the RenderDevice after the plugins are built to support webgpu.

https://bevyengine.org/learn/migration-guides/0.10-0.11/#webgpu-support

**Reply by dragostis** [reply]
Brilliant answer! Thank you so much.
